### DISCUSSION TITLE
Using a rust crate from crates.io

### DISCUSSION THREAD

**Discussion by RayyanAnsari** [fzyzcjy/flutter_rust_bridge#1305] [Q&A]
Hi,
I've read the docs for integrating rust code within a flutter app, however, they all create their own rust crate and write code into `lib.rs` and `api.rs`. All I want to do is use the functions from an existing rust crate published on crates.io - how do I do this?

**Comment by fzyzcjy** [comment]
I guess maybe https://cjycode.com/flutter_rust_bridge/feature/lang_external.html#types-in-other-crates?

And feel free to PR to improve the automation that makes you write even less code for mirroring a crate!

**Reply by RayyanAnsari** [reply]
I tried to use flutter_rust_bridge_codegen on the library's lib.rs and I got this output:
```
Error: Failed to parse function output type `HashMap<PluginID,PluginInfo>`
```

**Reply by fzyzcjy** [reply]
Map is not supported yet, but should be easy to PR by the following approach: Just mimic how a `String` is implemented (delegate its implementation to `Vec<u8>`), we can handle `HashMap<K,V>` by delegating to `Vec<(K,V)>`
